wildfire_one.py

```diff
@@ -0,0 +1,67 @@
+""" This module contains methods for retrieving information from the WFWX Fireweather API.
+"""
+import logging
+from aiohttp import ClientSession, BasicAuth
+from schemas import WeatherStation
+import config
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def get_auth_token(session):
+    """ Get an auth token for WFWX Fireweather API
+    """
+    password = config.get('WFWX_SECRET')
+    user = config.get('WFWX_USER')
+    auth_url = config.get('WFWX_AUTH_URL')
+    async with session.get(auth_url, auth=BasicAuth(login=user, password=password)) as response:
+        return await response.json()
+
+
+def is_station_valid(station):
+    """ Run through a set of conditions to check if the station is valid.
+
+    Returns True if station is good, False is station is bad.
+    """
+    if station['stationStatus']['id'] != 'ACTIVE':
+        return False
+    if station['latitude'] is None or station['longitude'] is None:
+        return False
+    return True
+
+
+def de_serialize_station(station):
+    """ Transform from the json object returned by wf1, to our station object.
+    """
+    return WeatherStation(
+        code=station['stationCode'],
+        name=station['displayLabel'],
+        lat=station['latitude'],
+        long=station['longitude'])
+
+
+async def get_stations():
+    """ Get list of stations from WFWX Fireweather API
+    """
+    max_page_size = config.get('WFWX_MAX_PAGE_SIZE')
+    base_url = config.get('WFWX_BASE_URL')
+    async with ClientSession() as session:
+        token = await get_auth_token(session)
+        headers = {'Authorization': 'Bearer {}'.format(token['access_token'])}
+        total_pages = 1
+        page = 0
+        stations = []
+        while page < total_pages:
+            url = '{base_url}/v1/stations?size={size}&sort=displayLabel&page={page}'.format(
+                base_url=base_url, size=max_page_size, page=page)
+            LOGGER.debug('loading station page %d...', page)
+            async with session.get(url, headers=headers) as response:
+                station_json = await response.json()
+                LOGGER.debug('done loading station page %d.', page)
+            total_pages = station_json['page']['totalPages']
+            for station in station_json['_embedded']['stations']:
+                if is_station_valid(station):
+                    stations.append(de_serialize_station(station))
+            page = page + 1
+        LOGGER.debug('total stations: %d', len(stations))
+    return stations
```

Comment:
Can you turn this into smaller chunks?(i.e. `fetch_stations(url)` & `parse_stations(stations)`) it seems a bit imperative to me

Reply:
I've changed this section of code significantly... please have another look.